jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/HttpStreamOverFCGI.java

```diff
@@ -101,15 +102,17 @@ else if (FCGI.Headers.QUERY_STRING.equalsIgnoreCase(name))
             _query = value;
         else if (FCGI.Headers.SERVER_PROTOCOL.equalsIgnoreCase(name))
             _version = value;
+        else if (FCGI.Headers.HTTPS.equalsIgnoreCase(name))
+            _secure = value;
         else
             processField(field);
     }
 
     public void onHeaders()
     {
         String pathQuery = URIUtil.addPathQuery(_path, _query);
-        // TODO https?
-        MetaData.Request request = new MetaData.Request(_method, HttpScheme.HTTP.asString(), hostPort, pathQuery, HttpVersion.fromString(_version), _headers, -1);
+        HttpScheme scheme = _secure != null ? HttpScheme.HTTPS : HttpScheme.HTTP;
```

Comment:
RFC 3875 (https://datatracker.ietf.org/doc/html/rfc3875) does not define `HTTPS`, but apparently for PHP it must be set to a non-empty value to indicate that the request was secure (https://www.php.net/manual/en/reserved.variables.server.php).

We use `HTTPS=on` for compatibility with PHP applications such as WordPress, but other applications require `true` or `1`, see for example: https://build.trac.wordpress.org/browser/branches/6.2/wp-includes/load.php#L1473

So `HTTPS=false` as per PHP spec would still mean that the request is secure because it's non-empty, but it really depends on the application.

🤷🏼

Reply:
Maybe we should have a list of affirmative values: "true", "1", "on", "secure", "https". Anything else is not secure.

At the very least can we exclude the empty string?